jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1616,6 +1616,114 @@ ecma_op_typedarray_define_index_prop (ecma_object_t *obj_p, /**< a TypedArray ob
   return ECMA_VALUE_TRUE;
 } /* ecma_op_typedarray_define_index_prop */
 
+/**
+ * Specify the creation of a new TypedArray
+ * object using a constructor function.
+ *
+ * See also: ES11 22.2.4.6
+ *
+ * Used by:
+ *        - ecma_typedarray_species_create
+ *
+ * @return ecma_value_t function object from created from constructor_p argument
+ *
+ */
+ecma_value_t ecma_typedarray_create (ecma_object_t *constructor_p, /**< constructor function */
+                                     ecma_value_t *arguments_list_p, /**< argument list */
+                                     uint32_t arguments_list_len) /**< length of argument list */
+{
+  ecma_value_t ret_val = ecma_op_function_construct (constructor_p,
+                                                     constructor_p,
+                                                     arguments_list_p,
+                                                     arguments_list_len);
+  if (ECMA_IS_VALUE_ERROR (ret_val))
+  {
+    return ret_val;
+  }
+
+  if (!ecma_is_typedarray (ret_val))
+  {
+    ecma_free_value (ret_val);
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Constructed object is not TypedArray."));
+  }
+
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (ret_val);
+
+  if ((arguments_list_len == 1) && (ecma_is_value_number (arguments_list_p[0])
+                                    || ecma_is_value_bigint (arguments_list_p[0])))
```

Comment:
Are you sure that `ecma_is_value_bigint (arguments_list_p[0])` is required here? the following `ecma_op_to_number ` check is going to throw error instantly for bigints. According to the standard only a single number check is required here 
 followed by a `get_number_value` instead of a `to_number` conversion